src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,150 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.exception.LoadException;
+import duke.exception.RonException;
+import duke.exception.WriteException;
+import duke.parser.StorageParser;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Stores tasks from the chatbot in an ArrayList for easy retrieval
+ */
+public class TaskList {
+    private List<Task> store;
+    private final File file;
+
+    public TaskList(File file) throws RonException {
+        this.file = file;
+        this.store = new ArrayList<>();
+        try {
+            Scanner fr = new Scanner(file);
+            while (fr.hasNextLine()) {
+                String currTask = fr.nextLine();
+                String[] tokens = currTask.split("\\| ");
+                if (tokens[0].charAt(0) == "T".charAt(0)) {
+                    Task task = new Todo("todo " + tokens[2]);
+                    if (Integer.parseInt(tokens[1].trim()) == 1) {
+                        task.toggleStatus();
+                    }
+                    store.add(task);
+                }
+                if (tokens[0].charAt(0) == "D".charAt(0)) {
+                    Task task = new Deadline("deadline " + tokens[2] + "/by " + tokens[3]);
+                    if (Integer.parseInt(tokens[1].trim()) == 1) {
+                        task.toggleStatus();
+                    }
+                    store.add(task);
+                }
+                if (tokens[0].charAt(0) == "E".charAt(0)) {
+                    Task task = new Event("event " + tokens[2] + "/at " + tokens[3]);
+                    if (Integer.parseInt(tokens[1].trim()) == 1) {
+                        task.toggleStatus();
+                    }
+                    store.add(task);
+                }
+            }
+            fr.close();
+        } catch (IOException e) {
+            throw new LoadException();
+        }
+    }
+
+    public TaskList(String filePath) {
+        this.file = new File(filePath);
+        this.store = new ArrayList<>();
+
+        if (!file.getParentFile().exists()) {
+            file.getParentFile().mkdirs();
+        }
+    }
+
+    /**
+     * Retrieves task from given index
+     *
+     * @param index
+     * @return Task in given index
+     */
+    public Task get(int index) {
+        return this.store.get(index);
+    }
+
+    public void printTasks() {
+        int i = 1;
```

Comment:
Do you think it will be clearer if `i` is given a better name? Likewise in `findTasks`! Also, I really love your coding style 😍 

```suggestion
        int taskId = 1;
```